Samples/PDESampleApplication/Form2.cs

```diff
@@ -0,0 +1,319 @@
+                var result = await g_udpm.UnprotectBufferAsync(protectedBuffer);
+                if (result.Status == UserDataBufferUnprotectStatus.Succeeded)
+                {
+                    LogLine("Result of Unprotecting the buffer:" +
+                        CryptographicBuffer.ConvertBinaryToString(BinaryStringEncoding.Utf8, result.UnprotectedBuffer));
+                    LogLine("Status of Unprotectng the buffer:" + result.Status);
+                }
+                else
+                {
+                    LogLine("This protected buffer is currently unavailable for unprotection");
+                }
+            }
+            catch(ArgumentException)
+            {
+                LogLine("Invalid Argument, please check text being unprotected!");
+            }
+            catch (Exception)
+            {
+                LogLine("Something went wrong");
+            }
+        }
+
+        async void ProtectBuffer(String text, UserDataAvailability level)
+        {
+            try
+            {
+                var buffer = CryptographicBuffer.ConvertStringToBinary(text, BinaryStringEncoding.Utf8);
+                g_protectedContent = await g_udpm.ProtectBufferAsync(buffer, level);
+                g_protectbase64EncodedContent = CryptographicBuffer.EncodeToBase64String(g_protectedContent);
+                LogLine("Protected buffer: " + g_protectbase64EncodedContent);
+            }
+            catch (ArgumentException)
+            {
+                LogLine("Invalid Argument, please check text being protected!");
+            }
+            catch (Exception)
+            {
+                LogLine("Something went wrong");
+            }
+        }
+        private void button4_Click(object sender, EventArgs e)
+        {
+            if(folderBrowserDialog1.ShowDialog() == DialogResult.OK)
+            {
+                listView1.Items.Add(folderBrowserDialog1.SelectedPath.Trim());
+                string selectedPath = folderBrowserDialog1.SelectedPath;
+                if (File.Exists(selectedPath))
+                {
+                    g_selectedFile = selectedPath;
+                } 
+                else if (Directory.Exists(selectedPath))
+                {
+                    g_selectedFolder = selectedPath; 
+                } else
+                {
+                    MessageBox.Show("Selected path is neither a file nor a directory please rectify");
+                }
+            } 
+        }
+
+        private void button5_Click(object sender, EventArgs e)
+        {
+            if (openFileDialog1.ShowDialog() == DialogResult.OK)
+            {
+                listView2.Items.Add(openFileDialog1.FileName.Trim());
+                g_selectedFile = openFileDialog1.FileName;
+                if (!File.Exists(g_selectedFile))
+                {
+                    MessageBox.Show("Selected file is not a file. please rectify");
```

Comment:
Same feedback here - let's not handle errors we don't expect. If the user selected the file, it should exist.